SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialRPCService.cpp

```diff
@@ -15,25 +15,27 @@ namespace SpatialGDK
 SpatialRPCService::SpatialRPCService(ExtractRPCDelegate ExtractRPCCallback, const USpatialStaticComponentView* View)
 	: ExtractRPCCallback(ExtractRPCCallback)
 	, View(View)
+	, FailureOutcomesToQueue(EPushRPCResult::None)
 {
 }
 
 EPushRPCResult SpatialRPCService::PushRPC(Worker_EntityId EntityId, ERPCType Type, RPCPayload Payload)
 {
 	EntityRPCType EntityType = EntityRPCType(EntityId, Type);
 
-	if (RPCRingBufferUtils::ShouldQueueOverflowed(Type) && OverflowedRPCs.Contains(EntityType))
+	if (QueuedRPCs.Contains(EntityType))
 	{
 		// Already has queued RPCs of this type, queue until those are pushed.
-		AddOverflowedRPC(EntityType, MoveTemp(Payload));
-		return EPushRPCResult::QueueOverflowed;
+		AddQueuedRPC(EntityType, MoveTemp(Payload));
+		return PushRPCResultUtils::MakeFailureResultCode(EPushRPCResult::AlreadyQueued, Type, FailureOutcomesToQueue);
 	}
 
-	EPushRPCResult Result = PushRPCInternal(EntityId, Type, MoveTemp(Payload));
+	const EPushRPCResult Result = PushRPCInternal(EntityId, Type, MoveTemp(Payload));
+	VALIDATE_RPC_PUSH_RESULT(Result);
```

Comment:
Should this be wrapped in a check/verify ?

Reply:
yes!  intended to put that in the macro

Reply:
Validate now actually validates.